collect_app/src/main/java/org/odk/collect/android/formlists/blankformlist/BlankFormListViewModel.kt

```diff
@@ -224,6 +207,7 @@ class BlankFormListViewModel(
             1 -> _allForms.value.sortedByDescending { it.formName.lowercase() }
             2 -> _allForms.value.sortedByDescending { it.dateOfCreation }
             3 -> _allForms.value.sortedBy { it.dateOfCreation }
+            4 -> _allForms.value.sortedByDescending { it.dateOfLastUsage }
```

Comment:
I think a really nice change we could make while we're in here would be to remove any knowledge of the display order of these sort options from `BlankFormListViewModel`. One thing that comes to mind is that we could have an `enum class` of sort "types" and an additional property on `FormListSortingOption` of that. Then, `FormListSortingAdapter`'s listener could receive that enum instead of the index which would allow this `when` to match on the sort type rather than the index.

I think a change like the above would make the code and tests easier to read as it wouldn't require any context of the display order.

Reply:
Ohh this would require a lot of changes and I'm not sure if it is worth it...
`FormListSortingAdapter` is supposed to be a class for all types of lists and different types of lists have different sorting options I think representing them by enums would require mixing all of them in one class.

I would need to have more time to warm up to this idea but I'm off tomorrow and on Friday so won't have time to do that. If you are sure this would make the code better please take over.

Reply:
Good points. Sounds like something we can come back to if we're in here again. No need to rush to block this at the moment.